src/components/entity/ha-chart-base.html

```diff
@@ -0,0 +1,514 @@
+<link rel='import' href='../../../bower_components/polymer/polymer.html'>
+
+<dom-module id="ha-chart-base">
+  <link rel="import" type="css" href="../../../bower_components/billboard.js/dist/billboard.min.css">
+  <template>
+    <style>
+       :host {
+        display: block;
+        /* @apply --my-custom-chart-istic-mixin; */
+      }
+      .tooltip {
+        position: absolute;
+        padding: 5px;
+        background: white;
+        border: 1px solid #AAA;
+        border-radius: 2px;
+        pointer-events: none;
+        z-index: 32767;
+      }
+      #chartTarget {
+        position: relative;
+      }
+    </style>
+    <div id="chartTarget"></div>
+  </template>
+  <script src="../../../bower_components/d3/d3.min.js"></script>
+  <script src="../../../bower_components/billboard.js/dist/billboard.min.js"></script>
+
+</dom-module>
+<script>
+/* global d3 bb */
+  /**
+ * Original: d3-gtimeline at Github
+ * Modified and rewrited for Home Assistant
```

Comment:
Why was it modified and rewritten? Would it be possible to contribute back to the original library the functionality that we need?

Reply:
Some feature can push back to the original one, but the data structure is changed. I moved the chart code to standalone Github repo.